### PR TITLE
Addressing issue #117

### PR DESCRIPTION
**Pull request by bdorney**
## Description
Addressing https://github.com/cms-gem-daq-project/vfatqc-python-scripts/issues/117.

`run_scans.py` will now properly pass the `--shelf` argument to the `preCmd` when the `--config` option is used.

### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

See above.  The above issue would prevent scans from being taken with the correct configuration.

## How Has This Been Tested?

Tests performed here: http://cmsonline.cern.ch/cms-elog/1005139

And analyzed here: http://cmsonline.cern.ch/cms-elog/1005447

The `--shelf=3` option was properly passed with calling the `--config` option when using commit: 56476318321bb01d254094c9631cd69285659935.

### Screenshots (if appropriate):
summary_vt1_20
fitsummary_vt1_20

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by bdorney** [comment]
The quality check:

> code-quality/landscape — Waiting for status to be reported

Seems to be stuck at this state since yesterday.  Any ideas as to why?

**Comment by jsturdy** [comment]
Hmm, not sure... I see on the [dashboard](https://landscape.io/github/cms-gem-daq-project/vfatqc-python-scripts/pulls/) that they have run fine...

**Review by jsturdy — CHANGES_REQUESTED** [review]
Looking at the code, the reason that the `preCmd` was not populated at the beginning was because not all tools would be expecting it, e.g., `ultraThreshold`, `ultraLatency`, `fastLatency`
This is to prevent running a configuration for a system that had previously been configured.  Best would be to revert that change and then evaluate whether the operational case needs `confChamber` for any type of tool that `run_scans` will execute

**Comment by bdorney** [comment]
Changes reverted, but shelf functionality still present.

For `ultraLatency.py` and `fastLatency.py` offhand I would expect it would be necessary to change VT1 and trim configuration for a given scan of interest.   On further investigation it seems these two scripts take some parameters but only use a few of them (e.g. they both take `--vt1` argument but they don't ever use it).  While they do use *some* parameters they are passed (e.g. `--vt2` in the case of `ultraLatency.py`).

Additionally for `ultraThreshold.py` you might want be interested in scanning with different mspl values.  But it does not appear to be used.

I think this highlights a level of disorganization we have inserted.  I would propose we move forward in a different way:

`run_scans.py` should set the configuration for each scan using the `--config` flag and `preCmd` list.  Then the scan application which is launched by `run_scans.py` should not do any configuration outside what is necessary for the scan.  For example the following scans appear to do extra configuration:
- `ultraLatency.py` sets `vt2`, `mspl`
- `ultraThreshold.py` sets `vt2`, `latency`
- `fastLatency.py` sets `mspl`

For easier maintenance and organization we should have `run_scans.py` call `confChamber.py` for the link of interest with the desired config and then the scan application should run.

Suggest to make this a new issue and include in a separate PR.

File: run_scans.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def launchTestsArgs(tool, shelf, slot, link, chamber, vfatmask, scanmin, scanmax
 
   #Build Commands
   setupCmds = []
-  preCmd = None
+  preCmd = None
```
Comment (jsturdy) [inline run_scans.py]: there appears to be some trailing whitespace here (L24)

File: run_scans.py [hunk 2]
```diff
@@ -132,7 +132,8 @@ def launchTestsArgs(tool, shelf, slot, link, chamber, vfatmask, scanmin, scanmax
       pass
     log = file("%s/scanLog.log"%(dirPath),"w")
     if preCmd and config:
-      runCommand(preCmd,log)
+      #runCommand(preCmd,log)
+      runCommand(preCmd)
```
Comment (jsturdy) [inline run_scans.py]: Any reason why you removed the logging (or just commented it out)?

**Review by jsturdy — CHANGES_REQUESTED** [review]

**Comment by jsturdy** [comment]
Right, when all update functionality has been added, we can then start to reorganize and package

**Comment by bdorney** [comment]
Testing updates: http://cmsonline.cern.ch/cms-elog/1005667

Also as a by-product when using the `--debug` the `vfatmask` will now be correctly written in hex.  The reason it was not being written as hex previously seems to be some weird interaction of the `dict.values()` method.  Calling `chamber_vfatmask.values()` returns a `list` where all the entries are formatted as `int`'s in base 10.  Using the `hex()` function and python list comprehension was able to solve this.

File: run_scans.py (hunk 1 above)
Comment (jsturdy) [reply run_scans.py]: Still there...

File: run_scans.py (hunk 2 above)
Comment (jsturdy) [reply run_scans.py]: ?

**Review by jsturdy — CHANGES_REQUESTED** [review]

File: run_scans.py (hunk 2 above)
Comment (jsturdy) [reply run_scans.py]: Removing the logging, is not the correct thing to do here.  If there's a problem, the problem should be addressed.
Logging in this manner has been working for a long time

**Review by jsturdy — CHANGES_REQUESTED** [review]

File: run_scans.py (hunk 1 above)
Comment (bdorney) [reply run_scans.py]: Searching for `\s` characters in vim.  Doesn't show up:

screen shot 2017-08-31 at 11 19 24

Comment (bdorney) [reply run_scans.py]: It may be that changes weren't pushed to the remote...?

Comment (jsturdy) [reply run_scans.py]: Agree, current file on github appears fixed

File: run_scans.py (hunk 2 above)
Comment (jsturdy) [reply run_scans.py]: Still needs resolution

**Review by jsturdy — CHANGES_REQUESTED** [review]

File: run_scans.py (hunk 2 above)
Comment (jsturdy) [reply run_scans.py]: NVM, see update now

**Review by jsturdy — APPROVED** [review]

**Comment by bdorney** [comment]
Looks like again `code-quality/landscape` is stuck in `Waiting for status to be reported` for ~9 days now...this seems to be a re-occurring issue.

**Pull request merged**